main/init/sbt/src/mill/main/sbt/SbtBuildGenMain.scala

```diff
@@ -0,0 +1,507 @@
+package mill.main.sbt
+
+import mainargs.{ParserForClass, arg, main}
+import mill.constants.Util
+import mill.main.buildgen.*
+import mill.main.buildgen.BuildGenUtil.*
+import mill.main.buildgen.IrDependencyType.*
+import os.Path
+
+import scala.collection.immutable.SortedSet
+
+/**
+ * Converts an sbt build to Mill by generating Mill build file(s).
+ * The implementation uses the sbt
+ * [[https://www.scala-sbt.org/1.x/docs/Combined+Pages.html#addPluginSbtFile+command addPluginSbtFile command]]
+ * to add a plugin and a task to extract the settings for a project using a custom model.
+ *
+ * The generated output should be considered scaffolding and will likely require edits to complete conversion.
+ *
+ * ===Capabilities===
+ * The conversion
+ *  - handles deeply nested modules
+ *  - captures publish settings
+ *  - configures dependencies for configurations:
+ *    - no configuration
+ *    - Compile
+ *    - Test
+ *    - Runtime
+ *    - Provided
+ *    - Optional
+ *  - configures testing frameworks (@see [[mill.scalalib.TestModule]]):
+ *    - Java:
+ *      - JUnit 4
+ *      - JUnit 5
+ *      - TestNG
+ *    - Scala:
+ *      - ScalaTest
+ *      - Specs2
+ *      - µTest
+ *      - MUnit
+ *      - Weaver
+ *      - ZIOTest
+ *      - ScalaCheck
+ * ===Limitations===
+ * The conversion does not support:
+ *  - custom dependency configurations
+ *  - custom settings including custom tasks
+ *  - sources other than Scala on JVM and Java, such as Scala.js and Scala Native
+ *  - cross builds
+ */
+@mill.api.internal
+object SbtBuildGenMain
+    extends BuildGenBase[Project, String, (BuildInfo, Tree[Node[Option[Project]]])] {
+  override type C = Config
+
+  def main(args: Array[String]): Unit = {
+    val cfg = ParserForClass[Config].constructOrExit(args.toSeq)
+    run(cfg)
+  }
+
+  private def run(cfg: Config): Unit = {
+    val workspace = os.pwd
+
+    println("converting sbt build")
+
+    def systemSbtExists(sbt: String) =
+      // The return code is somehow 1 instead of 0.
+      os.call((sbt, "--help"), check = false).exitCode == 1
+
+    val isWindows = Util.isWindows
+    val sbtExecutable = if (isWindows) {
+      val systemSbt = "sbt.bat"
+      if (systemSbtExists(systemSbt))
+        systemSbt
+      else
+        throw new RuntimeException(s"No system-wide `$systemSbt` found")
+    } else {
+      val systemSbt = "sbt"
+      // resolve the sbt executable
+      // https://raw.githubusercontent.com/paulp/sbt-extras/master/sbt
+      if (os.exists(workspace / "sbt"))
+        "./sbt"
+      else if (os.exists(workspace / "sbtx"))
+        "./sbtx"
```

Comment:
An alternative would be to add a `customSbt` config argument.

Reply:
Added and tested manually. Thanks.